tools/src/py-list-deps.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+
+# Copyright 2020, 2022 Datawire. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+# This script looks at Python3 source files in a directory, and parses
+# them to return a list of distributions that those Python sources
+# depend on.
+#
+# A quick refresher on PEP 561 terminology:
+#
+# - "distribution": a bundle that gets shared around (eggs and
+#   wheels are implementations of "distribution")
+#
+# - "module": a thing you can `import` (could be implemented either as
+#   `MOD.py` or as `MOD/__init__.py`)
+#
+# - "package": a module that contains more modules within it (a
+#   `MOD/__init__.py`)
+
+# This script started life as a script that generates Bazel 'BUILD'
+# files, and was more-or-less a clone of
+# <https://github.com/tuomasr/pazel>.  Why we wrote it, instead of
+# just using Pazel:
+#
+#  1. Pazel doesn't actually work; with Python 3.8 on GNU/Linux, the
+#     _is_in_stdlib function raises an exception
+#  2. Pazel doesn't understand relative imports.
+#  3. Pazel is too configurable; the configuration is complex enough
+#     that it's simpler/easier/faster to just have a purpose-built
+#     script that you can modify.
+#
+# And a good thing too; having a purpose-built script that we can
+# modify meant that we can leverage it to be useful without Bazel.
+
+import argparse
+import ast
+import importlib
+import os
+import sys
+import sysconfig
+from typing import List, NamedTuple, Optional, Set
+
+
+def parse_members(filepath: str) -> Set[str]:
+    with open(filepath, 'r') as filehandle:
+        filecontent = filehandle.read()
+
+    with open(filepath, 'r') as filehandle:
+        filecontent = filehandle.read()
```

Comment:
One of these should probably go away, right?

Reply:
...actually, wait, why do we have `parse_members` and `parse_imports`?

Reply:
> ...actually, wait, why do we have `parse_members` and `parse_imports`?

Good question.

Short answer: IDK, I wrote this code like 2 years ago

Long answer: After reading it, `parse_members` returns all of the things in a file; the names of all functions, the names of all variables, the names of re-exported things imported from elsewhere... well, not the names of all variables; like the comment says, it doesn't support all statement types. But it "should", it's just that variables aren't implemented.